### PR TITLE
[curl] Add http3 + remove ssl + openssl default ssl + remove other ssl/tls backend features.

### PR DESCRIPTION
**Pull request by talregev**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/curl/use_ca_native.patch [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+index 656aa7c740facf..c4bea84462679a 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -399,6 +399,11 @@ cmake_dependent_option(CURL_USE_BEARSSL "Enable BearSSL for SSL/TLS" OFF CURL_EN
+ cmake_dependent_option(CURL_USE_WOLFSSL "Enable wolfSSL for SSL/TLS" OFF CURL_ENABLE_SSL OFF)
+ cmake_dependent_option(CURL_USE_GNUTLS "Enable GnuTLS for SSL/TLS" OFF CURL_ENABLE_SSL OFF)
+ 
++option(CURL_USE_CA_NATIVE "Use standard certificate store of operating system" OFF)
++if(CURL_USE_CA_NATIVE)
++  set(USE_CA_NATIVE ON)
```
Comment (Osyotr) [inline ports/curl/use_ca_native.patch:11]: No. Use `CURLSSLOPT_NATIVE_CA` runtime option.

Comment (talregev) [reply ports/curl/use_ca_native.patch:11]: ok. I will remove this feature.

File: ports/curl/vcpkg.json [hunk 2]
```diff
@@ -142,35 +159,6 @@
         "libssh2"
       ]
     },
-    "ssl": {
```
Comment (Osyotr) [inline ports/curl/vcpkg.json:145]: I don't see a reason why `ssl` should be removed and `openssl` made the default. Could you explain?

Comment (talregev) [reply ports/curl/vcpkg.json:145]: You can read all comments in my other PR start with @BillyONeal 
https://github.com/microsoft/vcpkg/pull/37146#discussion_r1520801099

**Comment by talregev** [comment]
@BillyONeal Do you want me to remove schannel and other ssl backend feature as well?

**Comment by dg0yt** [comment]
i.e. changing windows and osx backend to openssl.
Backend comparision: https://curl.se/docs/ssl-compared.html
- no automatic CRL
- no native cert check 
- extra deployment burden

alll for experimental http3 support.

**Comment by talregev** [comment]
> i.e. changing windows and osx backend to openssl. Backend comparision: https://curl.se/docs/ssl-compared.html
> 
> * no automatic CRL
> * no native cert check
> * extra deployment burden
> 
> alll for experimental http3 support.

You can enable native cert check, also for openssl. I wanted to enable it by default. I can make a patch for that.

**Comment by talregev** [comment]
> i.e. changing windows and osx backend to openssl. Backend comparision: https://curl.se/docs/ssl-compared.html
> 
> * no automatic CRL
> * no native cert check
> * extra deployment burden
> 
> alll for experimental http3 support.

Is this PR, or http3 with restriction, or without http3.

**Comment by dg0yt** [comment]
> Do you want me to remove schannel and other ssl backend features as well?

As long as other backends exist, you have multi-ssl. And cannot have http3.

**Comment by talregev** [comment]
> > Do you want me to remove schannel and other ssl backend features as well?
> 
> As long as other backends exist, you have multi-ssl. And cannot have http3.

As you wish.

File: ports/curl/use_ca_native.patch (hunk 1 above)
Comment (talregev) [reply ports/curl/use_ca_native.patch]: @BillyONeal I am thinking to bring the native certificate os by default. That even on windows, openssl can take native certificate from os. 
What do you think?
https://github.com/curl/curl/pull/13111

**Comment by BillyONeal** [comment]
Correct, the expectation was that on `osx | ios`, the TLS backend selected is `sectransp`, on `(windows & !uwp) | mingw` the backend selected, and `openssl` selected otherwise.

Does http3 support only work with openssl?

**Comment by talregev** [comment]
This document write down all the options for curl with http3.
https://curl.se/docs/http3.html

**Comment by BillyONeal** [comment]
> This document write down all the options for curl with http3. https://curl.se/docs/http3.html

Hmmm... that's unfortunate. Adding some maintainers of https://github.com/microsoft/msquic

@nibanks @rzikm @csujedihy Do we have guidance for how normal Windows folks should get to http3 that complies with machine policy?

**Comment by BillyONeal** [comment]
:sigh: I'm sorry for making the suggestion to do this, I didn't realize that removing the TLS features would make the platform default TLS backends non functional. I'm no longer sure this is the correct tradeoff. I asked some Windows folks about their thoughts...

**Comment by talregev** [comment]
@BillyONeal Don't be sorry, These PRs for that we check, test and learn. I am happy that we check all the options for http3.
My original offer to this PR is that openssl will check native certificate by default, include windows.
Then you can compile curl tool and test the new http3 connection.
I will make another commit later, with the command to test the curl http3 connection on windows (or any other os).

**Comment by talregev** [comment]
@BillyONeal I added the feature ca-native. also add it by default. 
On windows you can try:
```
vcpkg install curl[tool,http3]
cd installed\x64-windows\tools\curl
curl --http3 https://curl.se/ -I
```

Try and let me know what do you think.
We can also go back to my previous PR that is on draft: #37146
Or if there is other option.
Also we can wait for maintainers of msquic for their respond.

**Comment by BillyONeal** [comment]
Closing this as per https://github.com/microsoft/vcpkg/pull/37146#issuecomment-2026153067 ; sorry our misunderstanding sent you on this path :/

**Pull request closed without merging**